tests/NationalInstruments.Analyzers.UnitTests/ChainOfMethodsWithLambdasAnalyzerTests.cs

```diff
@@ -0,0 +1,648 @@
+            .Distinct();|>
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+}, // inside a property getter
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(IEnumerable<ISoftwareContent> softwareContent)
+        {
+            softwareContent.Select(content => <|content.Children.GroupBy(software => software.AliasName).Select(g => g.First()).Distinct()|>);
+        }
+
+        private interface ISoftwareContent
+        {
+            IEnumerable<ISoftwareContent> Children { get; }
+
+            string AliasName { get; }
+        }
+    }
+}
+"
+} // inside an argument
+            };
+
+        [Fact]
+        public void NI1017_NoLambdas_NoDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(ISoftwareContent softwareContent)
+        {
+            softwareContent.Child().ToString().TrimEnd();
+            softwareContent.Child().AliasName.TrimEnd().Length.ToString();
+        }
+
+        private interface ISoftwareContent
+        {
+            ISoftwareContent Child();
+
+            string AliasName { get; }
+        }
+    }
+}
+");
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_OneLambda_NoDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(ISoftwareContent softwareContent)
+        {
+            softwareContent.Children().Any(s => s.AliasName == ""T"");
+            softwareContent.Children(x => true).Any();
+            softwareContent.FirstChild.Children().First().AliasName.Any(s => s == 'A');
+            softwareContent.FirstChild.Children(x => true).First().AliasName.Any();
+        }
+
+        private interface ISoftwareContent
+        {
+            ISoftwareContent FirstChild { get; }
+
+            IEnumerable<ISoftwareContent> Children(Predicate<int> predicate = null);
+
+            string AliasName { get; }
+        }
+    }
+}
+");
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_PoorlySplitMultipleLambdasInMethodInvocations_EmitsDiagnostic()
+        {
+            var test = new AutoTestFile(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(ISoftwareContent softwareContent)
+        {
+            <|softwareContent.Children(x => true).Any(s => s.AliasName == ""T"")|>;
```

Comment:
Since you don't actually use the captured text (stuff that occurs between `<|` and `|>`), you can just place `<|>` at the location of the diagnostic (`<|` in this case). Using the captured text means your diagnostic message to the user would include dynamic text.

Reply:
Done.